### PR TITLE
Update geometry_utilities.h

### PR DESCRIPTION
**Pull request by sunethwarna**
This PR adds 2D surface node extraction to ShapeOptimizationApplication/geometry_utilities.

### CODE REVIEW

File: applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h [hunk 1]
```diff
@@ -147,26 +147,51 @@ class GeometryUtilities
...
+                }
+            }
+            else if (domain_size == 2)
+            {
+                Element::GeometryType::GeometriesArrayType edges = itElem->GetGeometry().Edges();
+
+                for(unsigned int edge=0; edge<edges.size(); edge++)
+                {
+                    // Create vector that stores all node is of current face
+                    vector<unsigned int> ids(edges[edge].size());
```
Comment (loumalouomega) [inline applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h]: Use DenseVector instead

File: applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h [hunk 2]
```diff
@@ -147,26 +147,51 @@ class GeometryUtilities
...
+                Element::GeometryType::GeometriesArrayType faces = itElem->GetGeometry().Faces();
 
-    		for(unsigned int face=0; face<faces.size(); face++)
-    		{
-    			// Create vector that stores all node is of current face
-    			vector<unsigned int> ids(faces[face].size());
+                for(unsigned int face=0; face<faces.size(); face++)
+                {
+                    // Create vector that stores all node is of current face
+                    vector<unsigned int> ids(faces[face].size());
```
Comment (loumalouomega) [inline applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h]: Idem

**Comment by sunethwarna** [comment]
@loumalouomega done :)

File: applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h [hunk 3]
```diff
@@ -147,26 +147,51 @@ class GeometryUtilities
...
 
+        unsigned int domain_size = static_cast<unsigned int>(mrModelPart.GetProcessInfo()[DOMAIN_SIZE]);
+
     	// Fill map that counts number of faces for given set of nodes
     	for (ModelPart::ElementIterator itElem = mrModelPart.ElementsBegin(); itElem != mrModelPart.ElementsEnd(); itElem++)
     	{
-    		Element::GeometryType::GeometriesArrayType faces = itElem->GetGeometry().Faces();
+            if (domain_size==3)
+            {
+                Element::GeometryType::GeometriesArrayType faces = itElem->GetGeometry().Faces();
```
Comment (armingeiser) [inline applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h]: i think you could reduce the code:

```cpp
Element::GeometryType::GeometriesArrayType child_geometries; //find a better name
if (domain_size==3)
    child_geometries=itElem->GetGeometry().Faces();
else
    child_geometries=itElem->GetGeometry().Edges();
```

the rest is the same for both versions. If you just replace the word faces/edges with something more general. e.g. ```childs``` or something similar

**Review by armingeiser — COMMENTED** [review]
@dbaumgaertner can you think of a better name for this function if it now also works for 2D and extracts the EdgeNodes?

File: applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h (hunk 3 above)
Comment (sunethwarna) [reply applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h]: @armingeiser how about ExtractBoundaryNodes ?

Comment (armingeiser) [reply applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h]: Sounds good!

**Comment by sunethwarna** [comment]
@armingeiser done

**Review by armingeiser — APPROVED** [review]
Thanks!

**Pull request merged**